provider/semgrep/index.ts

```diff
@@ -0,0 +1,116 @@
+import type {
+    Annotation,
+    AnnotationsParams,
+    AnnotationsResult,
+    ItemsParams,
+    ItemsResult,
+    MentionsParams,
+    MentionsResult,
+    MetaParams,
+    MetaResult,
+    Provider,
+} from '@openctx/provider'
+
+import { type Finding, type Findings, urlfor } from './api.js'
+import API from './client.js'
+
+export type Settings = {
+    deployment: string // Semgrep deployment slug
+    token: string // Semgrep app token
+    repo?: string // Semgrep scan repository
+}
+
+function parseUri(link: string): { deployment: string; finding: number } | null {
+    const url = new URL(link)
+    if (url.hostname.includes('semgrep.dev')) {
+        try {
+            const [d, f] = url.pathname.split('/orgs/')[1].split('/findings/')
+            return { deployment: d, finding: parseInt(f) }
+        } catch (err) {
+            return null
+        }
+    }
+    return null
+}
+
+function aiPrompt(finding: Finding): string {
+    return [
+        finding.rule_name ?? '',
+        finding.rule_message ?? '',
+        finding.rule.name ?? '',
+        finding.rule.message ?? '',
+        finding.triage_comment ?? '',
+        finding.triage_reason ?? '',
+        finding.assistant?.autofix?.fix_code ?? '',
+        finding.assistant?.autofix?.explanation ?? '',
+        finding.assistant?.autotriage?.reason ?? '',
+    ].join('\n')
+}
```

Comment:
not exactly sure if this is the right way to pass a prompt to `ai` in the `items()` method. are there any guides? looking for feedback.

Reply:
I see that for many providers we used XML, but we don't seem to be super consistent here.
Might be worth a shot? Look for 'fast-xml-parser'

Reply:
ok, I've also gone with the XML structure purely because it looks more intuitive than just exposing a bunch of raw text, but perhaps we should add some examples somewhere in the docs for `AssistantInfo.content?` property might look like?

/cc @sqs